### PR TITLE
Use pixel ratio to scale mouse events on HiDpi monitors

### LINKED ISSUE #4239: [apps] Rectangle selection tool does not work properly on retina display
**Context**

I run `pcl_point_cloud_editor` on a MacBook with 2880x1800 display resolution. The rectangle selection tool works in principle, but the selection does not scale properly with the mouse movement.
I suspect this has to do with HiDpi handling in Qt or OpenGL. I tried to dig around a bit, but am too unfamiliar with either framework to find the problem.

**Expected behavior**

When dragging selection rectangle, the corner should follow the mouse pointer.

**Current Behavior**

When dragging selection rectangle, the corner seems to follow the mouse pointer with some factor < 1 of its movement.

**To Reproduce**

1. Start `pcl_point_cloud_editor`
2. Load a PCD file
3. Select 2D selection tool
4. Drag around to observe incongruous selection.

**Screenshots/Code snippets**

editor

**Your Environment (please complete the following information):**

 - OS: MacOS 10.14.6
 - Compiler: Apple LLVM version 10.0.1 (clang-1001.0.46.4)
 - PCL Version: e6d032a1eed0115542ae9895679b75a5fca06eef

**Additional context**

I have inspected `QApplication::desktop()->screen()->size()`, which shows a resolution of 1920x1200 for my screen, which is 1/1.5 of the actual resolution. Maybe these are device-dependent pixels, which some Qt documentation hints at in regards to retina displays.

I attempted to scale the values processed in `mouse*Event()` callbacks in the main editor widget by 1.5, but this did not change the behaviour.

### PR DESCRIPTION
**Pull request by themightyoarfish**
Fixes #4239

It says [here](https://stackoverflow.com/a/39467033/2397253) that one should use `devicePixelRatio()` to scale everything that uses pixel values, such as mouse click events. I don't quite get the [documentation](https://doc.qt.io/qt-5/highdpi.html), but it seems to do the trick here.

Could somebody with a Mac or HiDpi display test this?

### CODE REVIEW

**Comment by themightyoarfish** [comment]
So could this be merged? Anything I can/should do to advance this PR?

**Comment by kunaltyagi** [comment]
It needs someone to test the program on normal and HiDpi monitor and confirm there's no regression.

**Comment by themightyoarfish** [comment]
The display I test this on has a resolution of 2880x1800, which I guess is High dpi (macbook retina display). When I move this program to a secondary display with 1440 x 2560, the scaling is off again, which i assume is because of different densities and the Qt app is not informed of the current screen, but whatever it was launched on?
Not sure if this is expected behaviour.

**Comment by themightyoarfish** [comment]
Fixed. Turns out one can simply query the `devicePixelRatio()` for any widget.

File: apps/point_cloud_editor/src/cloudEditorWidget.cpp [hunk 1]
```diff
@@ -41,6 +41,8 @@
 #include <QFileDialog>
 #include <QMessageBox>
 #include <QMouseEvent>
+#include <QApplication>
```
Comment (larshg) [inline apps/point_cloud_editor/src/cloudEditorWidget.cpp:44]: I guess these includes are not necessary?

**Comment by larshg** [comment]
I have tried tested this on various screens I thought would be highdpi, but unfortunately not. Every time it returns 1 in scale, so its not broken normal screens.
According to the doc, it would return 2 on highdpi screens, but that seems odd that this should do it. Unless they actually double the number of pixels in both height and width direction.
But I'm leaning towards its fine until proven otherwise on highDPI screens.

**Review by mvieth — APPROVED** [review]
I can't really reproduce the original error (on Ubuntu), nor can I confirm that the PR fixes it, but this looks like a plausible fix, so I am okay with merging it.

> According to the doc, it would return 2 on highdpi screens, but that seems odd that this should do it. Unless they actually double the number of pixels in both height and width direction

You mean [this documentation](https://doc.qt.io/qt-5/qscreen.html#devicePixelRatio-prop)? Sounds like it would return 2 on some retina displays, maybe those that have ~200 PPI? And would return values > 2 on displays that have even more PPI?

**Pull request merged**